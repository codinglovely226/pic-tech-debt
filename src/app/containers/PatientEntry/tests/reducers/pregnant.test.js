/* eslint-disable */

import PregnantReducer from '../../reducers/pregnant';

import {
  updatePatientPregnantLactating,
  updatePatientPregnantStatus,
  updatePatientPregnantTrimester,
  updatePatientPregnantTrimesterUnknown,
} from '../../actions/case';

describe('PregnantReducer', () => {
  it('returns the initial state', () => {
    expect(PregnantReducer(undefined, {})).toMatchSnapshot();
  });
  it('update lactating', () => {
    expect(PregnantReducer({}, updatePatientPregnantLactating(1, false, true))).toMatchSnapshot();
  });
  it('update status', () => {
    expect(PregnantReducer({}, updatePatientPregnantStatus(1, 'Old Status', 'New Status'))).toMatchSnapshot();
  });
  it('update trimester', () => {
    expect(PregnantReducer({}, updatePatientPregnantTrimester(1, 1, 2))).toMatchSnapshot();
  });
  it('update trimesterUnknown', () => {
    expect(PregnantReducer({}, updatePatientPregnantTrimesterUnknown(1, true, false))).toMatchSnapshot();
  });
});
