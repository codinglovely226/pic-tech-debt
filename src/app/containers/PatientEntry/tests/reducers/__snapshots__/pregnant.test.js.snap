exports[`PregnantReducer returns the initial state 1`] = `
Object {
  "lactating": "unknown",
  "status": "unknown",
  "trimester": "unknown",
  "trimesterUnknown": null,
}
`;

exports[`PregnantReducer update lactating 1`] = `
Object {
  "lactating": true,
  "status": "unknown",
  "trimester": "unknown",
  "trimesterUnknown": null,
}
`;

exports[`PregnantReducer update status 1`] = `
Object {
  "lactating": "unknown",
  "status": "New Status",
  "trimester": "unknown",
  "trimesterUnknown": null,
}
`;

exports[`PregnantReducer update trimester 1`] = `
Object {
  "lactating": "unknown",
  "status": "unknown",
  "trimester": 2,
  "trimesterUnknown": null,
}
`;

exports[`PregnantReducer update trimesterUnknown 1`] = `
Object {
  "lactating": "unknown",
  "status": "unknown",
  "trimester": "unknown",
  "trimesterUnknown": false,
}
`;
